internal/apiclient/eventstreams.go

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2023 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiclient
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/hyperledger/firefly-transaction-manager/pkg/apitypes"
+)
+
+func (c *fftmClient) GetEventStreams(ctx context.Context) ([]apitypes.EventStream, error) {
+	resp, err := c.client.R().
+		SetContext(ctx).
+		SetResult([]apitypes.EventStream{}).
```

Comment:
Can't you create an object like `var result []apitypes.EventStream` and then pass `SetResult(&result)` to have it automatically unmarshalled? Up to you if that's cleaner or not, but I think that's what we've done in other places to avoid having to do an explicit cast later.

Reply:
Yes. But that definitely didn't work when I tried it before 🙃